packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1111,6 +1118,13 @@ class WidgetsApp extends StatefulWidget {
   /// {@endtemplate}
   final String? restorationScopeId;
 
+  /// If true, only builds a [MediaQuery] if none is available above the
+  /// [WidgetsApp].
+  ///
+  /// If set to false [WidgetsApp] build a [MediaQuery] using
+  /// [MediaQuery.fromWindow].
+  final bool useExistingMediaQuery;
```

Comment:
WDYT of `useInheritedMediaQuery` or `buildMediaQueryFromWindow` (which would revers the logic here) or something like that?

Reply:
Sure, went with `useInheritedMediaQuery`